src/utils/time-utils.ts

```diff
@@ -0,0 +1,107 @@
+import {
+    CAPTURE_INTERVAL_RE,
+    DURATION_RE,
+    POSTGRES_INTERVAL_RE,
+} from 'validation';
+import { hasLength } from './misc-utils';
+
+interface SegmentedInterval {
+    h: number;
+    m: number;
+    s: number;
+}
+
+const parsePostgresInterval = (interval: string): SegmentedInterval => {
+    const [hours, minutes, seconds] = interval.split(':').map((segment) => {
+        const numericSegment = Number(segment);
+
+        return isFinite(numericSegment) ? numericSegment : -1;
+    });
+
+    return { h: hours, m: minutes, s: seconds };
+};
+
+const parseCaptureInterval = (interval: string): SegmentedInterval => {
+    let hours = -1;
+    let minutes = -1;
+    let seconds = -1;
+
+    interval.split(' ').forEach((segment) => {
+        const numericSegment = Number(segment.substring(0, segment.length - 1));
+        const numericValue = isFinite(numericSegment) ? numericSegment : -1;
+
+        if (segment.includes('h')) {
+            hours = numericValue;
+        } else if (segment.includes('m')) {
+            minutes = numericValue;
+        } else if (segment.includes('s')) {
+            seconds = numericValue;
+        }
```

Comment:
We might want to check `endsWith` to be really explicit.

Reply:
Given this function is only consumed after a string is confirmed to take the capture interval shape, this _should_ be fine. There is a minor concern about the regular expression as it isn't as tight as I would prefer (e.g., `"2h  "` is a valid interval), but the change handlers _should_ trim leading and trailing spaces appropriately.